test/unit-tests/expression/function/evaluate.test.js

```diff
@@ -84,4 +84,101 @@ describe('evaluate', function () {
+      assert.strictEqual(math.evaluate('"" ?? 42'), '')
+      assert(isNaN(math.evaluate('NaN ?? 42'))) // Should return NaN, not 42
+    })
+
+    it('should handle nullish coalescing with variables', function () {
+      const scope1 = {}
+      assert.throws(() => math.evaluate('x ?? 42', scope1), /Undefined symbol x/)
+
+      const scope2 = { x: null }
+      assert.strictEqual(math.evaluate('x ?? 42', scope2), 42)
+
+      const scope3 = { x: 0 }
+      assert.strictEqual(math.evaluate('x ?? 42', scope3), 0)
+
+      const scope4 = { x: undefined }
+      assert.strictEqual(math.evaluate('x ?? 42', scope4), 42)
+
+      const scope5 = { x: 5 }
+      assert.strictEqual(math.evaluate('x ?? 42', scope5), 5)
+    })
+
+    it('should handle chained nullish coalescing', function () {
+      assert.strictEqual(math.evaluate('null ?? undefined ?? 42'), 42)
+      assert.strictEqual(math.evaluate('null ?? 10 ?? 42'), 10)
+      assert.strictEqual(math.evaluate('5 ?? null ?? 42'), 5)
+      assert.strictEqual(math.evaluate('null ?? null ?? null ?? 99'), 99)
+    })
+
+    it('should handle nullish coalescing with correct precedence', function () {
+      // ?? has lower precedence than arithmetic operators
+      assert.strictEqual(math.evaluate('null ?? 1 + 2'), 3) // null ?? (1 + 2)
+      assert.strictEqual(math.evaluate('null ?? 2 * 3'), 6) // null ?? (2 * 3)
+      assert.strictEqual(math.evaluate('null ?? 2^3'), 8) // null ?? (2^3)
+
+      // ?? has same precedence as logical OR, left associative
+      assert.strictEqual(math.evaluate('null ?? false or true'), true) // (null ?? false) or true
+      assert.strictEqual(math.evaluate('true or null ?? 42'), true) // (true or null) ?? 42 = true ?? 42 = true
+
+      // Parentheses can override precedence
+      assert.strictEqual(math.evaluate('null ?? (1 + 2)'), 3)
+      assert.strictEqual(math.evaluate('(null ?? 1) + 2'), 3) // (null ?? 1) + 2 = 1 + 2 = 3
+    })
+
+    it('should handle nullish coalescing with complex expressions', function () {
+      const scope = { a: null, b: 5, c: 0 }
+      assert.strictEqual(math.evaluate('a ?? b * 2', scope), 10) // null ?? (5 * 2)
+      assert.strictEqual(math.evaluate('c ?? b * 2', scope), 0) // 0 ?? (5 * 2) = 0
+      assert.strictEqual(math.evaluate('(a ?? b) * 2', scope), 10) // (null ?? 5) * 2 = 10
+    })
+
+    it('should handle nullish coalescing with strings', function () {
+      assert.strictEqual(math.evaluate('null ?? "hello"'), 'hello')
+      assert.strictEqual(math.evaluate('"world" ?? "hello"'), 'world')
+      assert.strictEqual(math.evaluate('"" ?? "hello"'), '') // empty string is not nullish
+    })
+
+    it('should handle nullish coalescing with matrices and arrays', function () {
+      assert.deepStrictEqual(math.evaluate(['null ?? 1', '2 ?? null', 'null ?? null ?? 3']), [1, 2, 3])
+      assert.deepStrictEqual(math.evaluate(math.matrix(['null ?? 1', '2 ?? null'])), math.matrix([1, 2]))
+    })
```

Comment:
These tests only test a matrix with entries given by nullish coalescing. Please add tests that show the (proper) behavior of the nullish coalescing operator when the left, right, or both operands are themselves arrays or matrices. Thanks.